Hyper not pinned to 0.14.14

I'm on Rust 1.56, installed tonic on my existing project and got the following error:

````
error[E0277]: the trait bound `hyper::common::exec::Exec: hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not satisfied
   --> C:\Users\gcpth\.cargo\registry\src\github.com-1ecc6299db9ec823\tonic-0.6.0\src\transport\server\mod.rs:510:18
    |
510 |                 .with_graceful_shutdown(signal)
    |                  ^^^^^^^^^^^^^^^^^^^^^^ the trait `hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not implemented for `hyper::common::exec::Exec`
    |
    = help: the following implementations were found:
              <hyper::common::exec::Exec as hyper::common::exec::NewSvcExec<I, N, S, E, W>>

error[E0277]: `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)` cannot be shared between threads safely
   --> C:\Users\gcpth\.cargo\registry\src\github.com-1ecc6299db9ec823\tonic-0.6.0\src\transport\server\mod.rs:510:18
    |
510 |                 .with_graceful_shutdown(signal)
    |                  ^^^^^^^^^^^^^^^^^^^^^^ `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)`
    = note: required because of the requirements on the impl of `Sync` for `Unique<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>`
    = note: required because it appears within the type `Box<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>`
    = note: required because it appears within the type `Pin<Box<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>>`
    = note: required because it appears within the type `UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>`

error[E0277]: the trait bound `hyper::common::exec::Exec: hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not satisfied
   --> C:\Users\gcpth\.cargo\registry\src\github.com-1ecc6299db9ec823\tonic-0.6.0\src\transport\server\mod.rs:508:13
    |
508 | /             server
509 | |                 .serve(svc)
510 | |                 .with_graceful_shutdown(signal)
511 | |                 .await
    | |______________________^ the trait `hyper::common::exec::NewSvcExec<ServerIo<IO>, futures_util::future::Ready<Result<tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, tower::util::BoxService<http::Request<hyper::Body>, http::Response<UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>>, Box<(dyn std::error::Error + Send + Sync + 'static)>>, hyper::common::exec::Exec, hyper::server::shutdown::GracefulWatcher>` is not implemented for `hyper::common::exec::Exec`
    |
    = help: the following implementations were found:
              <hyper::common::exec::Exec as hyper::common::exec::NewSvcExec<I, N, S, E, W>>
    = note: required because of the requirements on the impl of `futures_core::Future` for `hyper::server::shutdown::Graceful<impl Accept, MakeSvc<<L as Layer<S>>::Service, IO>, F, hyper::common::exec::Exec>`
note: required by `poll`
   --> C:\Users\gcpth\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\future\future.rs:99:5
    |
99  |     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)` cannot be shared between threads safely
   --> C:\Users\gcpth\.cargo\registry\src\github.com-1ecc6299db9ec823\tonic-0.6.0\src\transport\server\mod.rs:508:13
    |
508 | /             server
509 | |                 .serve(svc)
510 | |                 .with_graceful_shutdown(signal)
511 | |                 .await
    | |______________________^ `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)`
    = note: required because of the requirements on the impl of `Sync` for `Unique<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>`
    = note: required because it appears within the type `Box<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>`
    = note: required because it appears within the type `Pin<Box<(dyn HttpBody<Data = bytes::Bytes, Error = Box<(dyn std::error::Error + Send + Sync + 'static)>> + Send + 'static)>>`
    = note: required because it appears within the type `UnsyncBoxBody<bytes::Bytes, Box<(dyn std::error::Error + Send + Sync + 'static)>>`
    = note: required because of the requirements on the impl of `futures_core::Future` for `hyper::server::shutdown::Graceful<impl Accept, MakeSvc<<L as Layer<S>>::Service, IO>, F, hyper::common::exec::Exec>`
note: required by `poll`
   --> C:\Users\gcpth\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\future\future.rs:99:5
    |
99  |     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0277`.
error: could not compile `tonic` due to 4 previous errors
````

I cannot reproduce this error with a new project with only tonic installed. It seems to only happen in my project.

The build stops at tonic (it does not even get to my code). Can anyone point me in the right direction on how to resolve?

--- a/tonic/Cargo.toml
+++ b/tonic/Cargo.toml
@@ -71,7 +71,7 @@ async-trait = {version = "0.1.13", optional = true}
 
 # transport
 h2 = {version = "0.3", optional = true}
-hyper = {version = "0.14.4", features = ["full"], optional = true}
+hyper = {version = "0.14.14", features = ["full"], optional = true}
 hyper-timeout = {version = "0.4", optional = true}
 tokio = {version = "1.0.1", features = ["net"], optional = true}
 tokio-stream = "0.1"

